File: CS50 Problem Sets/Problem Set 6/sentimental/readability.py
from cs50 import get_string

text = get_string("Text: ")

letters = 0
words = 0
sentences = 0
digits = 0
special = 0

for i in text:
    # if alphabet
    if i.isalpha():
        letters += 1
    # if numeric
    elif i.isnumeric():
        digits += 1
    # if space
    elif i == ' ':
        words += 1
    # if pointe or etcetre
    elif i == "." or i == "!" or i == "?":
        sentences += 1
    else:
        special += 1

# Get L - the average number of letters per 100 words in the text
L = float(letters / words * 100)

# Get S - the average number of sentences per 100 words in the text
S = float(sentences / words * 100)

# Readbility
index = round((float)(0.0588 * L - 0.296 * S - 15.8))


# Output
if index < 1:
    print("Before Grade 1")

elif index >= 16:
    print("Grade 16+")

else:
    print("Grade ", index)

File: CS50 Problem Sets/Problem Set 2/readability/readability.c
#include <cs50.h>
#include <stdio.h>
#include <string.h>
#include <ctype.h>
#include <math.h>

int count_letters(string text);
int count_words(string text);
int count_sentences(string text);



int main(void)
{
    string text = get_string("Text: ");

    int letters = count_letters(text);
    int words = count_words(text);
    int sentences = count_sentences(text);

    float L = ((float) letters / (float) words) * 100;
    float S = ((float) sentences / (float) words) * 100;

    float index = 0.0588 * L - 0.296 * S - 15.8; // found grade
    int rounded = round(index); // round index

    // if big than 16
    if (rounded > 16)
    {
        printf("Grade 16+\n");
    }
    //if less than 1
    else if (rounded < 1)
    {
        printf("Before Grade 1\n");
    }
    // simple print
    else
    {
        printf("Grade %i\n", rounded);
    }



}






// Counting letters
int count_letters(string text)
{
    int count = 0;
    for (int i = 0; i < strlen(text); i++)
    {
        char c = text[i];
        if (isalpha(c) != 0)
        {
            count += 1;
        }
    }

    return count;
}


// Counting words
int count_words(string text)
{
    int count = 0;
    for (int i = 0; i < strlen(text); i++)
    {
        char c = text[i];
        if (isspace(c) != 0)
        {
            count++;
        }
    }
    return count + 1;

}



// Counting Sentences
int count_sentences(string text)
{
    int count = 0;
    for (int i = 0; i < strlen(text); i++)
    {
        char c = text[i];
        if (c == '.' || c == '!' || c == '?')
        {
            count++;
        }
    }
    return count;
}